### PR TITLE
Added GPG Guide

### PR DESCRIPTION
**Pull request by alopresto**
Normally we just commit directly to the repository, but I wanted to get some feedback on this guide. People who have little to no experience with GPG will provide the best feedback, as this guide is designed to help them get acclimated and started. Experienced GPG users are also welcome to provide feedback, as I expect they will have more advanced critiques.

### CODE REVIEW

File: src/pages/markdown/gpg.md [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+Are you sure that you want to sign this key with your
+key "Andy LoPresto <alopresto@apache.org>" (3C6EF65B2F7DEF69)
+
+Really sign? (y/N) y
+gpg: RSA/SHA512 signature from: "3C6EF65B2F7DEF69 Andy LoPresto <alopresto@apache.org>"
+```
+
+### Web of Trust
+
+If it is infeasible to directly contact the key bearer directly, you can delegate that trust to a third-party who you already trust. For example, if you cannot directly contact Joe Witt, but you already trust Andy LoPresto (i.e. you have verified Andy's key and believe Andy would verify keys he trusts), you can sign Joe's key if Andy has already done so. You can choose to employ a lower level of validity (`marginal` vs. `full`) in that case depending on your transitive **ownertrust** in Andy's verification. See [Web of Trust][web-of-trust] and [GNU Privacy Handbook][gnu-privacy-handbook] for more information and good diagrams.
```
Comment (alopresto) [inline src/pages/markdown/gpg.md]: Repeated "directly".

File: src/pages/markdown/gpg.md [hunk 2]
```diff
@@ -0,0 +1,524 @@
+---
+title: GPG (Gnu Privacy Guard / Open PGP) Signatures
+---
+
+# Apache NiFi Release Signatures and Code Signing
+
+The purpose of this document is to capture and describe the steps involved in generating and verifying cryptographic signatures of official releases of Apache NiFi, as well as configuring cryptographic signatures of individual code commits.  It is written for contributors, committers, and users of Apache NiFi (and related applications).  
+
+## <a name="table-of-contents">Table of Contents</a>
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: The following links are broken:

Set up a GPG key
Sign your GPG key
Sign an external key
Signing code

File: src/pages/markdown/gpg.md [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+  * [Variable Reference Substitutions](#variable-reference-substitutions)
+  * [Download GPG](#download-gpg)
+  * [Set up a GPG key](#set-up-a-gpg-key)
+  * [Sign your GPG key](#sign-your-gpg-key)
+  * [Publish your key](#publish-your-key)
+  * [Import another GPG key](#import-another-gpg-key)
+  * [Verify a key](#verify-a-key)
+  * [Sign an external key](#sign-an-external-key)
+  * [Web of trust](#web-of-trust)
+  * [Set up git with your key](#set-up-git-with-your-key)
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Capitalize "Git" for consistency

File: src/pages/markdown/gpg.md [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+  * [Signing a release artifact](#signing-a-release-artifact)
+  * [Verifying a release signature](#verifying-a-release-signature)
+
+## <a name="the-objective">The Objective</a>
+
+Our aim is to instruct users on how to sign their commits, verify other's signatures, and do the same for official releases of Apache NiFi. 
+  
+## <a name="background-material">Background Material</a>
+
+  - These documents are helpful for general environmental setup to perform GPG signing and signature verification
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Consider adding a colon at the end or adding more space before the bulleted list.

File: src/pages/markdown/gpg.md [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+uid           [ultimate] NiFi Test User <test@nifi.apache.org>
+sub   rsa2048 2018-07-10 [E] [expires: 2020-07-09]
+
+```
+
+You now have a key generated for your identity. Some Apache users will include "(CODE SIGNING KEY)" after their name to separate this key from other keys they use. This is optional. By default, keys use 2048 bit length. You can increase this to 4096 bits by using the `--full-generate-key` or `--default-new-key-algo rsa4096` flags when generating. 
+
+The **key fingerprint** can be referred to by the last 8 hex digits (**short ID**) or last 16 hex digits (**long ID**), so for the remainder of this guide, our example key will have the **long ID** `7145 6940 555D B64A` (spaces optional) and **short ID** `555D B64A`, which we can reference in commands as `0x555DB64A`.  
+
+### Trust vs. Validity
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Change Trust to Overtrust?

File: src/pages/markdown/gpg.md [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+
+It is important to understand the difference between two closely related concepts -- **ownertrust** and **validity**. Validity is how much *you* trust a *key*; that is *how well you have verified the key represents who it claims*. Ownertrust is how much transitive trust you give to that entity; how well do you believe *Person X* verifies keys that *they* sign? 
+
+There are four levels of **ownertrust** and five levels of **validity**. 
+
+**Trust**
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Change to Overtrust?

File: src/pages/markdown/gpg.md (hunk 1 above)
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Perhaps change "good diagrams" to "helpful diagrams".

File: src/pages/markdown/gpg.md [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+
+    NIFI-5399 Added wildcard certificate documentation to Admin Guide.
+    Clarified CN vs. SAN entries.
+
+    This closes #2870.
+
+    Signed-off-by: Pierre Villard <pierre.villard.fr@gmail.com>
+```
+
+See [git-sign-tag-instructs], [github-help-gpg], and [git-gpg] for more on setting this up.
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Change "for more" to "for more information".

File: src/pages/markdown/gpg.md [hunk 8]
```diff
@@ -0,0 +1,524 @@
...
+    Signed-off-by: Pierre Villard <pierre.villard.fr@gmail.com>
+```
+
+See [git-sign-tag-instructs], [github-help-gpg], and [git-gpg] for more on setting this up. 
+
+## <a name="verifying-a-signature">Verifying a signature</a>
+
+When viewing commits by other authors, each may have a signature. The validity of the signature depends on your trust of the signer's key. 
+
+*Note: sometimes, the __author__ and __signer__ of a commit are different, especially in NiFi's RTC context.* The *author* is who wrote the code/content in the change, while the *signer* is the *committer* who actually merged the code to the `master` branch.
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: Consider changing *Note:  to NOTE: to be consistent with the usage in line 78.

File: src/pages/markdown/gpg.md [hunk 9]
```diff
@@ -0,0 +1,524 @@
...
+
+    NIFI-5237:
+    - Ensuring the proxy headers are considered when redirecting the user following a OIDC or Knox login exchange.
+
+    This closes #2763.
+
+    Signed-off-by: Bryan Bende <bbende@apache.org>
+```
+
+See [GitHub Blog: GPG Signature Verification][github-gpg-signing] for more on setting this up.
```
Comment (andrewmlim) [inline src/pages/markdown/gpg.md]: "more" to "more information"

**Comment by andrewmlim** [comment]
Added comments for readability, formatting, broken links etc.

**Comment by alopresto** [comment]
Thanks @andrewmlim . I incorporated your feedback.

File: src/pages/markdown/gpg.md [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+    - [Git Ready: Signing Releases with GPG][git-sign-tag-instructs]
+    - [RFC 4880: IETF Standard Spec for OpenPGP][rfc-4880]
+    - [GitHub Blog: GPG Signature Verification][github-gpg-signing]
+    - [Git Ready: gpg-sign releases][git-sign-tag-instructs]
+    - [GitHub Help: Signing Commits Using GPG][github-help-gpg]
+    - [GitHub Help: Telling Git About Your GPG Key][git-config-gpg]
+    - [Git Docs: Git Tools Signing Your Work][git-gpg]
+    - [PGP Web of Trust][web-of-trust]
+
+## <a name="terms">Terms</a>
```
Comment (markobean) [inline src/pages/markdown/gpg.md]: Add definitions of "PGP" and "GPG" to terms. Useful on its own, but also these acronyms are so similar and can look like an inadvertent typo.

**Pull request closed without merging**